test/test_autograd.py

```diff
@@ -4697,6 +4697,13 @@ def fn(a, b):
         with self.assertRaisesRegex(RuntimeError, bad_mark_dirty_err):
             fn(a, b)
 
+    def test_named_tensor_for_complex_views(self):
+        names = ["batch", "height", "width", "complex"]
+        z = torch.ones((5, 12, 14, 2), requires_grad=True).refine_names(*names)
+        z_complex = torch.view_as_complex(z.rename(None)).refine_names(*names[:-1])
+        z_complex.sum().backward()
+        self.assertEqual(z.grad, torch.ones_like(z))
```

Comment:
`self.assertEqual(z.grad, torch.view_as_real(torch.ones_like(z_complex)))`

Reply:
This should then be it:

```
names = ["batch", "height", "width", "complex"]
z = torch.ones((5, 12, 14, 2), requires_grad=True)
z_named = z.refine_names(*names)
z_complex = torch.view_as_complex(z_named.rename(None)).refine_names(*names[:-1])
z_complex.sum().backward()
self.assertEqual(z.grad, torch.view_as_real(torch.ones_like(z_complex).rename(None)))
```

Test it in my branch with `torch.allclose`. Is true.